src/CoreFoundation/CFException.cs

```diff
@@ -107,31 +110,33 @@ public static CFException FromCFError (IntPtr cfErrorHandle)
 		public static CFException FromCFError (IntPtr cfErrorHandle, bool release)
 		{
 			if (cfErrorHandle == IntPtr.Zero)
-				throw new ArgumentNullException (nameof (cfErrorHandle));
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (cfErrorHandle));
 
 			var e = new CFException (
 					CFString.FromHandle (CFErrorCopyDescription (cfErrorHandle), releaseHandle: true),
-					(NSString) Runtime.GetNSObject (CFErrorGetDomain (cfErrorHandle)),
+					Runtime.GetNSObject (CFErrorGetDomain (cfErrorHandle)) as NSString,
 					CFErrorGetCode (cfErrorHandle),
 					CFString.FromHandle (CFErrorCopyFailureReason (cfErrorHandle), releaseHandle: true),
 					CFString.FromHandle (CFErrorCopyRecoverySuggestion (cfErrorHandle), releaseHandle: true));
 
 			var cfUserInfo = CFErrorCopyUserInfo (cfErrorHandle);
 			if (cfUserInfo != IntPtr.Zero) {
 				using (var userInfo = new NSDictionary (cfUserInfo)) {
-					foreach (var i in userInfo)
-						e.Data.Add (i.Key?.ToString (), i.Value?.ToString ());
+					foreach (var i in userInfo) {
+						if (i.Key is not null)
+							e.Data.Add (i.Key.ToString (), i.Value?.ToString ());
+					}
 				}
 			}
 			if (release)
 				CFObject.CFRelease (cfErrorHandle);
 			return e;
 		}
 
-		public nint Code {get; private set;}
-		public NSString Domain {get; private set;}
-		public string FailureReason {get; private set;}
-		public string RecoverySuggestion {get; private set;}
+		public nint? Code {get; private set;}
```

Comment:
You can't make this nullable, because it's a struct and that would be a breaking change.